mod/state-transition/pkg/core/state_processor_genesis_test.go

```diff
@@ -298,25 +320,25 @@ func commonChecksValidators(
 
 	idx, err := bs.ValidatorIndexByPubkey(dep.Pubkey)
 	require.NoError(t, err)
-	require.Equal(t, math.U64(dep.Index), idx)
 
 	val, err := bs.ValidatorByIndex(idx)
 	require.NoError(t, err)
 	require.Equal(t, dep.Pubkey, val.Pubkey)
 
 	var (
 		maxBalance = math.Gwei(cs.MaxEffectiveBalance())
-		minBalance = math.Gwei(cs.EffectiveBalanceIncrement())
+		increment  = math.Gwei(cs.EffectiveBalanceIncrement())
+		minBalance = math.Gwei(cs.EjectionBalance())
 	)
 	switch {
 	case dep.Amount >= maxBalance:
 		require.Equal(t, maxBalance, val.EffectiveBalance)
-	case dep.Amount >= minBalance && dep.Amount < maxBalance:
+	case dep.Amount > minBalance && dep.Amount < maxBalance:
```

Comment:
this is the correct check. Genesis validator whose balance is equal to EjectionBalance are immediately ejected